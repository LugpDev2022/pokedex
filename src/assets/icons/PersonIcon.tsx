interface Props {
  size?: number;
  className?: string;
}

export const PersonIcon = ({ size = 30, className = "" }: Props) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width={size}
    height={size}
    fill="currentColor"
    className={`bi bi-person-fill ${className}`}
    viewBox="0 0 16 16"
    data-testid='svg'
  >
    <path d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z" />
  </svg>
);
