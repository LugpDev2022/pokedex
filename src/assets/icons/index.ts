export * from "./GoogleIcon";
export * from "./PersonIcon";
export * from "./SearchIcon";
export * from "./OutlinedStarIcon";
