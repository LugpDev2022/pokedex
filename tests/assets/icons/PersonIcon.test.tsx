import React from "react";
import { render, screen } from "@testing-library/react";
import { PersonIcon } from "../../../src/assets/icons/PersonIcon";

describe("tests on <PersonIcon />", () => {
  test("should match snapshot", () => {
    const { container } = render(<PersonIcon />);
    expect(container).toMatchSnapshot();
  });

  test("should have the given size", () => {
    render(<PersonIcon size={75} />);

    const svg = screen.getByTestId("svg");
    expect(svg.getAttribute("width")).toBe("75");
    expect(svg.getAttribute("height")).toBe("75");
  });

  test("should have the given size", () => {
    render(<PersonIcon className="testing-class another-testing-class" />);

    const svg = screen.getByTestId("svg");
    expect(svg.classList.contains("testing-class")).toBeTruthy();
    expect(svg.classList.contains("another-testing-class")).toBeTruthy();
  });
});
