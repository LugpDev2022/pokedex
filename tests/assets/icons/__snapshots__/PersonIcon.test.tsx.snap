// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <PersonIcon /> should match snapshot 1`] = `
<div>
  <svg
    class="bi bi-person-fill "
    data-testid="svg"
    fill="currentColor"
    height="30"
    viewBox="0 0 16 16"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
    />
  </svg>
</div>
`;
